src/en/RoyalRoad.lua

```diff
@@ -15,23 +128,87 @@ local function expandURL(url)
 	return baseURL .. (url:sub(1, 1) == "/" and "" or "/") .. url
 end
 
+local function TriStateFilter_(int, str)
+	if TriStateFilter then
+		return TriStateFilter(int, str)
+	else
+		return CheckboxFilter(int, str)
+	end
+end
```

Comment:
This backwards compatibility hack is not needed.

Reply:
TriStateFilter_ is for forward compatibility. Not using it means no one can install royalroad from the time the pull request is merged to the time TriStateFilter is added to a new shosetsu release.
Using it means multi-genre include filters work now, with multi-genre exclude filters delayed until later.